producers/http/handlers.go

```diff
@@ -136,11 +136,10 @@ func containerAppMetricHandler(p *producerImpl) http.HandlerFunc {
 			producers.AppMetricPrefix, cid,
 		}, producers.MetricNamespaceSep)
 
-		appMetrics, ok := p.store.Get(key)
-		if !ok {
-			httpLog.Errorf("/v0/containers/{id}/app/{metric-id} - not found in store: %s", key)
-			http.Error(w, "Key not found in store", http.StatusNoContent)
-			return
+		appMetrics, err := p.store.GetByRegex(key + ".*" + mid)
```

Comment:
Is this regex expression going to handle finding any metrics whose qualified names have tags appended to them?  For example, previously we created the qualified metric name by doing:

```
qualifiedName = qualifiedName + producers.MetricNamespaceSep + k + producers.MetricNamespaceSep + v
```

Which means that the tag name/value would be appended.

The above regex `key + ".*" + mid` seems like it may not match them?  

Ultimately, I feel like we may need to using some other representation for the `Store` which basically maps "qualified name" -> metrics.  We may need a key that is a struct, instead of a string.